editor/src/components/editor/store/editor-state.ts

```diff
@@ -752,6 +752,54 @@ export function editorStateInspector(
   }
 }
 
+export interface DraggedImageProperties {
+  width: number
+  height: number
+  src: string
+}
+
+interface NotDragging {
+  type: 'NOT_DRAGGING'
+}
+
+interface DraggingFromFS {
+  type: 'DRAGGING_FROM_FS'
+}
+
+export interface DraggingFromSidebar {
+  type: 'DRAGGING_FROM_SIDEBAR'
+  draggedImageProperties: DraggedImageProperties
+}
+
+export type DragSessionState = NotDragging | DraggingFromFS | DraggingFromSidebar
```

Comment:
Please can we rename this to reflect the fact it is specifically for asset dragging (or just image dragging?)

Reply:
absolutely, renamed it